extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -1058,6 +1058,7 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
             this.element.getAttribute('data-amp-slot-index'));
         return Promise.reject('Cannot access body of friendly frame');
       }
+      setImportantStyles(this.element, {position: 'static'});
```

Comment:
This will cause a full reflow of the page, so it needs to be done in measure/mutate phases. 

Also, how does `position: absolute` affect the measurements? The only thing it should change is child elements will consider it the `offsetParent`, which is why we set all elements to absolute by default.

Reply:
Are you sure it will cause reflow? The elements affected here are guaranteed to have `height: 0px` at the time this code runs. 

The presence of `position: absolute` appears to make the iframe's body element think that it has more width available to it then it actually does. For example, on the page I've tested, there's a wrapping div that constrains the width to something like 2/3rds of the viewport. With `position: absolute` the iframe's body element appears to ignore this constraint, and its width reads as the full viewport. Without `position: absolute` its width is constrained by the wrapping div, as expected.

Reply:
Absolutely.

![reflow 2019-05-07 14_14_02](https://user-images.githubusercontent.com/112982/57322779-67e4c600-70d2-11e9-974e-36fb61a90676.gif)

Can you link to the test page?

Reply:
Sent instructions offline. Now that I think about it, maybe this can be resolved by just adding `position: static !important;` here: https://github.com/ampproject/amphtml/blob/master/extensions/amp-ad/0.1/amp-ad.css#L77.